app/services/master_controller.py

```
@@ -2388,6 +2409,220 @@ async def set_trading_mode(self, mode: str) -> Dict[str, Any]:
+                "base_trading_mode": current_mode.value,
+                "is_custom": False
+            }
+            
+        except Exception as e:
+            self.logger.error("Failed to get AI model weights", user_id=user_id, error=str(e))
+            return {
+                "success": False,
+                "error": str(e)
+            }
+    
+    async def emergency_stop(self, user_id: str, reason: str = "manual_stop") -> Dict[str, Any]:
```

Ruff (0.12.2)

Redefinition of unused `emergency_stop` from line 2337

(F811)